Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -86,17 +99,53 @@ public override string DataSource
             }
         }
 
-        public override string ServerVersion => _connectionState == ConnectionState.Open ? SfSession.serverVersion : "";
+        public override string ServerVersion => IsOpen() ? SfSession.serverVersion : "";
 
         public override ConnectionState State => _connectionState;
 
+        private object? GetTransactionId()
+        {
+            if (!IsOpen())
+            {
+                return DBNull.Value;
+            }
+            using (IDbCommand command = CreateCommand())
+            {
+                command.CommandText = "SELECT CURRENT_TRANSACTION()";
+                return command.ExecuteScalar(); 
+            }
+        }
+
+        public bool HasTransactionInProgress() => GetTransactionId() != DBNull.Value;
+        
+        private bool TerminateTransaction(object? transactionInProgressId)
+        {
+            if (transactionInProgressId == DBNull.Value)
+                return false;
+            try
+            {
+                using (IDbCommand command = CreateCommand())
+                {
+                    command.CommandText = "ROLLBACK";
+                    command.ExecuteNonQuery(); 
+                    logger.Warn("Rolled back transaction: " + transactionInProgressId + " in session: " + SfSession.sessionId);
+                    return true;
+                }
+            }
+            catch (SnowflakeDbException exception)
+            {
+                logger.Error("Unable to rollback transaction: " + transactionInProgressId + " in session: " + SfSession.sessionId + ", exception: " + exception.Message);

```

Comment:
should we pass exception to logging method? We are loosing stack trace now?

Reply:
We should see just the reason for not terminating transaction during "dirty" connection being closed.
Either way this operation succeeds (and connection gets its rollback and returns to the pool) or fails (connection gets destroyed with underlying http session and doesn't return to the pool.
We can care here for the stack trace but user should worry about that what was suggested in the ticket.

Reply:
On the other hand - logs without stack traces are good if this is the only place from which this message can be written (like logs specific for some domain logic etc). Now you have a very generic method which may be closed from multiple places, and based on the message itself user may have difficulties in locating which function/operation/query/whatever caused the error. If we have a stack trace here, it would be easier to know. What do you think?